integrations/pr_publish.py

```
@@ -0,0 +1,785 @@
+
+# --------------------------------------------------------------------------- cli
+
+def _resolve_repo(arg):
+    # Shared validation lives in _check_repo (also enforced inside GitHubClient); this adds the
+    # CLI-specific source resolution and error hint.
+    repo = arg or os.environ.get("GITHUB_REPOSITORY", "")
+    try:
+        return _check_repo(repo)
+    except ValueError:
+        raise ValueError("repo must be 'owner/name' with no path/query characters (got %r); "
+                         "set --repo or $GITHUB_REPOSITORY" % (repo,))
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)